### DISCUSSION TITLE
Where do I specify python requirement in pyproject.py

### DISCUSSION THREAD

**Discussion by venaturum** [python-poetry/poetry#3569] [Q&A]
I'm not sure if I understand how the python requirement should be specified in pyproject.toml

The problem I have is this

- Python 3.9 requires Pandas >= 1.1.3
- Pandas >= 1 does not support Python 3.6

If I put 

```
[tool.poetry.dependencies]
python = "^3.6"
pandas
```

in my pyproject file then an environment solve should not be possible.  However when I run "poetry install" with a python 3.9 then it does "solve" and I get the following installs:

```
Package operations: 5 installs, 0 updates, 0 removals

  • Installing six (1.15.0)
  • Installing numpy (1.19.5)
  • Installing python-dateutil (2.8.1)
  • Installing pytz (2020.5)
  • Installing pandas (0.25.3)
```

The pandas version is too small for Python 3.9, and automated testing with Travis CI fails as a result

If I try to force it by setting

```
[tool.poetry.dependencies]
python = "^3.6"
pandas="^1.1.3"
```

Then the install fails to solve, stating that pandas requires Python >= 3.7.1

I can remove the python requirement from the dependencies and that would solve the problem, but then how do I restrict users from trying to use my package with a python version less than 3.6?

**Comment by venaturum** [comment]
The following travis ci config solved the problem I was having.  If there is a simpler solution please let me know.

NB package is called mypkg, and has a directory in the root folder called "tests"

```
# Config file for automatic testing at travis-ci.com


language: python
python:
  - "3.6"
  - "3.7"
  - "3.8"
  - "3.9"  

before_install:
  - pip install poetry codecov
    
install:
  - cd ..
  - poetry new temp
  - cd temp
  - poetry add ../mypkg
  - poetry add pytest-cov
  - poetry add codecov
  
script:
  - poetry run pytest -v --color=yes --cov-report=xml --cov=mypkg ../mypkg/tests
  
after_success:
  - codecov
```

**Comment by sinoroc** [comment]
Maybe be more explicit on your `pandas` dependency requirement specification. You could add some preferred version ranges to help the dependency resolver.
